src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -182,6 +183,9 @@ static int add_cryptsetup(const char *id, const char *what, bool rw, bool requir
         }
 
         return 0;
+#else
+        return log_debug_errno(-EOPNOTSUPP, "Encrypted partition, please set -Dlibcryptsetup=true to support it");
```

Comment:
please replace `-EOPNOTSUPP` by `SYNTHETIC_ERRNO(EOPNOTSUPP)`

i.e. for error codes we propagate that we make up ourselves we use SYNTHETIC_ERRNO() which is a macro that marks the error code in a special way so that it is not included in the ERRNO field in structured logging. We want that ERRNO= is only set for errors we *receive*, not for those we generate